### PR TITLE
Connects to AgileVentures/MetPlus_tracker#210

### PR DESCRIPTION
**Pull request by patmbolger**
Added feature "Update Job Category"

### CODE REVIEW

File: app/assets/javascripts/agency_admin.js [hunk 1]
```diff
@@ -54,38 +54,83 @@ var AgencyData = {
...
+      paginate_url = '/agency_admin/job_properties?data_type=' +
+                     'job_categories&job_categories_page=1';
+    }
+    AgencyData.get_updated_data('#job_categories_table',
+                                paginate_url);
+    $(model_errors_id).html(''); // Clear model errors in modal
+    $(modal_id).modal('hide');
+  },
+  change_job_category_error: function (exception, xhrObj, model_errors_id) {
+    // If model error(s), show content in div in modal (Firefox seems to add a
```
Comment (joaopapereira) [inline app/assets/javascripts/agency_admin.js]: This looks like something that is repeatable in a large group of places maybe we should remove this function from this object and create a more generic error object

File: app/controllers/job_categories_controller.rb [hunk 2]
```diff
@@ -1,15 +1,39 @@
...
-                      locals: {object: jc}, status: 422
+                      locals: {object: category}, status: 422
+    end
+  end
+
+  def edit
+    begin
+      category = JobCategory.find(params[:id])
+    rescue
+      render nothing: true, status: 404
```
Comment (joaopapereira) [inline app/controllers/job_categories_controller.rb]: Does this please any error messages to be displayed after the Ajax request?

**Comment by joaopapereira** [comment]
Looks good please review my comments

**Comment by patmbolger** [comment]
I extracted a few generic functions into 'data_management.js'.  There may be more opportunities which I will look at when I add "Skills" to the page.

In the meantime, I will be working on job category "delete" feature.

**Comment by joaopapereira** [comment]
Looks good, going to merge

**Pull request merged**